### PR TITLE
Improve stats: add route table stats, swap histograms for summaries.

### PR DESCRIPTION
**Pull request by fasaxc**
Switch to monotime for calculating time deltas.  Removes need to check for time discontinuities.  Nice thing about that switch is the type system ensures that you can't get confused between a monotime timestamp (which is a `time.Duration`) and a `time.Time`.

### CODE REVIEW

File: routetable/route_table.go [hunk 1]
```diff
@@ -148,6 +165,8 @@ func (r *RouteTable) Apply() error {
 			}
 		}
 		r.inSync = true
+
+		listIfaceTime.Observe(time.Since(listStartTime).Seconds())
```
Comment (neiljerram) [inline routetable/route_table.go]: No longer worried about backwards time jumps?

(Perhaps you're happy now that they don't actually occur in this interface.  That would be great if so!)

Comment (fasaxc) [inline routetable/route_table.go]: I've switch to the monotime package in the most recent commit.

**Comment by fasaxc** [comment]
@neiljerram Back to you to review switch to `monotime`.

**Comment by fasaxc** [comment]
retest this please

**Pull request merged**